src/simulation.rs

```diff
@@ -53,6 +58,22 @@ pub struct SimulationResponse {
     pub return_data: Bytes,
 }
 
+#[derive(Debug, Clone, Serialize, Deserialize)]
+pub struct StatefulSimulationRequest {
+    #[serde(rename = "chainId")]
+    pub chain_id: u64,
+    #[serde(rename = "gasLimit")]
+    pub gas_limit: u64,
+    #[serde(rename = "blockNumber")]
+    pub block_number: Option<u64>,
+}
+
+#[derive(Debug, Clone, Serialize, Deserialize, PartialEq)]
+pub struct StatefulSimulationResponse {
+    #[serde(rename = "statefulSimulationId")]
+    pub stateful_simulation_id: u32,
```

Comment:
I think UUIDs could be nicer than an incrementing id int. It means that if in future we allow a DB to be hooked up and multiple simulators to be run, the API doesn't need to change. I also frankly think the URLs will look nicer. :D  What do you think?